### PR TITLE
added store search

### PR DESCRIPTION
**Pull request by gammison**

### CODE REVIEW

File: static/js/load_map.js [hunk 1]
```diff
@@ -75,4 +76,60 @@ function initialize() {
...
         });
     });
+
+    const stores = document.getElementById('Stores');
+    function getLocation() {
+        if (navigator.geolocation) {
+            return navigator.geolocation.getCurrentPosition(showPosition);
+        } else {
+            x.innerHTML = "Geolocation is not supported by this browser.";
+        }
```
Comment (alanhdu) [inline static/js/load_map.js]: Switching between four space and two space indent?

File: static/js/load_map.js [hunk 2]
```diff
@@ -75,4 +76,60 @@ function initialize() {
             markers.push(nextMarker);
         });
     });
+
+    const stores = document.getElementById('Stores');
+    function getLocation() {
+        if (navigator.geolocation) {
+            return navigator.geolocation.getCurrentPosition(showPosition);
+        } else {
+            x.innerHTML = "Geolocation is not supported by this browser.";
```
Comment (alanhdu) [inline static/js/load_map.js]: Switching between four space and two space indent?

File: static/js/load_map.js [hunk 3]
```diff
@@ -75,4 +76,60 @@ function initialize() {
...
+        }
+    }
+
+    function showPosition(position) {
+        x.innerHTML = "Latitude: " + position.coords.latitude +
+        "<br>Longitude: " + position.coords.longitude;
```
Comment (alanhdu) [inline static/js/load_map.js]: The indentation here is weird. Also, should you really be setting `innerHTML` by hand?

File: static/js/load_map.js [hunk 4]
```diff
@@ -75,4 +76,60 @@ function initialize() {
...
+      }
+    }
+
+    stores.addEventListener('click', event => {
+      const request = {
+        location: columbia,
+        radius: '100',
+        types: ['store']
+      };
+      placeService.nearbySearch(request, callback);
```
Comment (alanhdu) [inline static/js/load_map.js]: I would just inline the callback here as an arrow function.

File: static/js/load_map.js [hunk 5]
```diff
@@ -75,4 +76,60 @@ function initialize() {
...
+    function showPosition(position) {
+        x.innerHTML = "Latitude: " + position.coords.latitude +
+        "<br>Longitude: " + position.coords.longitude;
+    }
+
+    function callback(results, status) {
+      if (status == google.maps.places.PlacesServiceStatus.OK){
+        markers.forEach(marker => {
+            marker.setMap(null);
+        });
```
Comment (alanhdu) [inline static/js/load_map.js]: Is this considered good javascript style?

IMO, you should use normal `for` loops for stateful changes (like this) and only use functional things like `forEach` or `map` for data transformations. But I could be wrong here.

File: static/js/load_map.js [hunk 6]
```diff
@@ -75,4 +76,60 @@ function initialize() {
...
+        markers = [];
```
Comment (alanhdu) [inline static/js/load_map.js]: Hm... I'm a little uncomfortable with having `markers` be global mutable state, but I doubt it's worthwhile to re-architect at this point.

File: static/js/load_map.js (hunk 2 above)
Comment (alexander-yu) [inline static/js/load_map.js]: Yeah, should probably make that consistent -- we seem to be using different indentations so I'll add a `.jshintrc`.

File: static/js/load_map.js (hunk 3 above)
Comment (alexander-yu) [inline static/js/load_map.js]: Looks like it's a result of incomplete copy/paste from an example -- should be better to just return `position.coords`.

File: static/js/load_map.js (hunk 5 above)
Comment (alexander-yu) [inline static/js/load_map.js]: I haven't seen any style suggestions for when to use `for` vs `forEach`; I've actually been of the impression that `forEach` is for when a data transformation isn't necessarily the goal, whereas `map` is expressly for data transformations. Not entirely sure, though -- I'm not the most familiar with ES5/6 style.

File: static/js/load_map.js (hunk 4 above)
Comment (alexander-yu) [inline static/js/load_map.js]: 👍

File: static/js/load_map.js (hunk 6 above)
Comment (alexander-yu) [inline static/js/load_map.js]: Yeah fair enough; at the very least I can move it inside the `initialize` function so it's block-scoped to the function as opposed to being global.

**Pull request merged**